src/js/components/Button/Button.js

```diff
@@ -82,10 +83,13 @@ const Button = forwardRef(
     };
 
     let buttonIcon = icon;
+
+    if (icon && size) buttonIcon = cloneElement(buttonIcon, { size: 'medium' });
```

Comment:
Shouldn't it just be `{ size }`? Also, this prevents  caller from overriding the size of an icon. I wonder if we should only explicitly set the size of the icon if the icon doesn't already have a size prop set? Similar to how we check for color below.

Reply:
Now that we decided to have an impact on the icon size, it is cleaner to remove this addition. It was an exploration until now.